app/controllers/katello/api/v2/host_module_streams_controller.rb

```diff
@@ -22,7 +22,7 @@ def index
     def index_relation
       return HostAvailableModuleStream.upgradable([@host]) if params[:status] == HostAvailableModuleStream::UPGRADABLE
 
-      rel = @host.host_available_module_streams
+      rel = @host.host_available_module_streams.where(available_module_stream_id: find_available_module_stream_ids)
```

Comment:
I'd like to see a test (or expectation added to `test_index_relation`) that simulates multiples being returned by `host_available_module_streams` that fails without this change.

Reply:
@jeremylenz added